lib/useKResponsiveWindow/index.js

```diff
@@ -2,125 +2,119 @@ import { computed, onBeforeUnmount, onMounted, ref } from '@vue/composition-api'
+const windowIsLarge = computed(() => windowBreakpoint.value > 2);
+const windowIsMedium = computed(() => windowBreakpoint.value === 2);
+const windowIsSmall = computed(() => windowBreakpoint.value < 2);
 
-  /**
-   * Check if the smallest window dimension(width or height) is smaller than the specified dimension
-   * @param {Number} dimension in px
-   * @returns {Boolean}
-   */
-  const smallestWindowDimensionIsLessThan = dimension => {
-    return (
-      windowBreakpoint.value < 4 && Math.min(windowWidth.value, windowHeight.value) < dimension
-    );
-  };
+const orientationQuery = new MediaQuery('screen and (orientation: portrait)', event => {
+  updateOrientation(event.matches);
+  updateGutter();
+});
+const heightQuery = new MediaQuery('screen and (max-height: 600px)', event => {
+  windowIsShort.value = event.matches;
+});
 
-  /**
-   * Update window orientation
-   * @param {Boolean} portrait
-   */
-  const updateOrientation = portrait => {
-    windowIsPortrait.value = portrait;
-    windowIsLandscape.value = !windowIsPortrait.value;
-  };
+function updateGutter() {
+  if (windowIsSmall.value || smallestWindowDimensionIsLessThan(600)) {
+    windowGutter.value = 16;
+  } else {
+    windowGutter.value = 24;
+  }
+};
+
+/**
+ * Initialize window properties
+ */
+function initProps() {
+  orientationQuery.eventHandler(orientationQuery.mediaQueryList);
+  heightQuery.eventHandler(heightQuery.mediaQueryList);
+  // convert below -- >>> ?
+  // for (const widthQuery of widthQueries) {
+  //   if (widthQuery.mediaQueryList.matches) {
+  //     widthQuery.eventHandler(widthQuery.mediaQueryList);
+  //     break;
+  //   }
+  // }
+};
+
+/**
+ * Start listening to media query changes
+ */
+function startListening() {
+  orientationQuery.startListening();
+  heightQuery.startListening();
+  // // convert below -- >> ?
+  // for (const widthQuery of widthQueries) {
+  //   widthQuery.startListening();
+  // }
+};
+
+/**
+ * Stop listening to media query changes
+ */
+function stopListening() {
+  orientationQuery.stopListening();
+  heightQuery.stopListening();
+  // convert below -- >>> ?
+  // for (const widthQuery of widthQueries) {
+  //   widthQuery.stopListening();
+  // }
+};
 
+/**
+ * Check if the smallest window dimension(width or height) is smaller than the specified dimension
+ * @param {Number} dimension in px
+ * @returns {Boolean}
+ */
+function smallestWindowDimensionIsLessThan(dimension) {
+  return (
+    windowBreakpoint.value < 4 && Math.min(windowWidth.value, windowHeight.value) < dimension
+  );
+};
+
+/**
+ * Update window orientation
+ * @param {Boolean} portrait
+ */
+function updateOrientation(portrait) {
+  windowIsPortrait.value = portrait;
+  windowIsLandscape.value = !windowIsPortrait.value;
+};
+
+
+/**
+ * Export window properties
+ * @returns {Object} Object with window properties
+ */
+export default function useKResponsiveWindow() {
   onMounted(() => {
```

Comment:
The simplest way to achieve this, with everything defined in the global scope would be to just call `initProps()` and then `startListening()` in the module scope, so that as soon as this is parsed, we start listening to things.

The other way of doing this, would be to follow the useKWindowDimensions route, and keep track of usage: https://github.com/learningequality/kolibri-design-system/blob/develop/lib/useKWindowDimensions.js#L74

So you would only call initProps and startListening if it wasn't already in use, and only call stopListening if the usage count has dropped to 0.